lib/auth/permissions.go

```diff
@@ -297,18 +302,19 @@ func (a *authorizer) authorizeRemoteUser(ctx context.Context, u RemoteUser) (*Co
 		KubernetesGroups: kubeGroups,
 		KubernetesUsers:  kubeUsers,
 		TeleportCluster:  a.clusterName,
-		Expires:          time.Now().Add(ttl),
+		Expires:          expires,
 
 		// These fields are for routing and restrictions, safe to re-use from
 		// unmapped identity.
-		Usage:             u.Identity.Usage,
-		RouteToCluster:    u.Identity.RouteToCluster,
-		KubernetesCluster: u.Identity.KubernetesCluster,
-		RouteToApp:        u.Identity.RouteToApp,
-		RouteToDatabase:   u.Identity.RouteToDatabase,
-		MFAVerified:       u.Identity.MFAVerified,
-		ClientIP:          u.Identity.ClientIP,
-		PrivateKeyPolicy:  u.Identity.PrivateKeyPolicy,
+		Usage:                     u.Identity.Usage,
+		RouteToCluster:            u.Identity.RouteToCluster,
+		KubernetesCluster:         u.Identity.KubernetesCluster,
+		RouteToApp:                u.Identity.RouteToApp,
+		RouteToDatabase:           u.Identity.RouteToDatabase,
+		MFAVerified:               u.Identity.MFAVerified,
+		MFAVerifiedSessionExpires: mfaVerifiedSessionExpires,
```

Comment:
This is either going to be empty, or identical to `Expires` right?

If that's the case, do we need to set it here or can we rely on the existing `Expires` and `MFAVerified` fields?

Reply:
@zmb3 I'm not sure what I was thinking, I think this makes sense: b571efe990b1b2db089199f002f4d83075eee335

@tigrato I'm not seeing how that relates

Reply:
> From what I understood, @zmb3 was asking to remove PreviousIdentityExpires here since its value is either empty or equal to Expires, and re-use the value of Expires and MFAVerified to decide the PreviousIdentityExpires.

I think what was meant was: at the time @zmb3's original comment was written, [the logic for setting `MFAVerifiedSessionExpires`](https://github.com/gravitational/teleport/pull/18607/files/919aca472825265ea285559e2ec2fe544c4564aa#diff-a0d9859c8f6c722d6937c51e77f3db91423e15cfebc385111d891ef30cdf218fR292-R296) (now called `PreviousIdentityExpires`) meant that it would either be empty or just the value of `Expires`, so it didn't really need to be set here.

However I don't know what I was thinking when I wrote that. My understanding is that what we're doing here is converting a local cluster identity to an identity that original identity is allowed to take up on a remote cluster. 

What I'm doing between https://github.com/gravitational/teleport/commit/b571efe990b1b2db089199f002f4d83075eee335 + https://github.com/gravitational/teleport/commit/a26f201546be180648fceee39c15da2abe427c49 is copying the original local cluster identity's `PreviousIdentityExpires` trait over, while ensuring it doesn't exceed the max allowed TTL for this new remote cluster identity by passing it through `AdjustSessionTTL` (the same process that was already being done for `Expires`).